### PR TITLE
Increase diffmatchcpatch timeout

### PR DESCRIPTION
**Pull request by vmarkovtsev**
Fixes https://github.com/src-d/go-git/issues/1083

Signed-off-by: Vadim Markovtsev <vadim@sourced.tech>

I can add a test, but it requires running 100 goroutines under `-race` and increases Travis run time by 10 minutes.

### CODE REVIEW

File: utils/diff/diff.go [hunk 1]
```diff
@@ -16,6 +17,7 @@ import (
 // string into the dst string.
 func Do(src, dst string) (diffs []diffmatchpatch.Diff) {
 	dmp := diffmatchpatch.New()
+	dmp.DiffTimeout = time.Hour  // the default is time.Second which may be too little under heavy load
```
Comment (mcuadros) [inline utils/diff/diff.go]: Well, maybe we can do this an argument?

Maybe `DoWithTimeout`?

Comment (vmarkovtsev) [reply utils/diff/diff.go]: Then we should remove the current `Do` function, because the root of the problem is a too small default timeout. wdyt?

Comment (mcuadros) [reply utils/diff/diff.go]: Nope, first we need to preserve the API, and second, I am not saying that we are going to keep the second, just that should be configurable.

Comment (vmarkovtsev) [reply utils/diff/diff.go]: So I should add `DoWithTimeout` and also preserve the changed default timeout, correct?

Comment (mcuadros) [reply utils/diff/diff.go]: Yup

**Comment by vmarkovtsev** [comment]
@mcuadros Done

**Pull request merged**